main.py

```diff
@@ -110,9 +118,67 @@ def load_model(checkpoint_path: str):
     return model
 
 
-def run(is_train: bool, model_path: str, strategy: dict):
-    if is_train:
-        for epoch in range(start_epoch, start_epoch + 55):
-            train(epoch)
-            test(epoch, model_path)
-            scheduler.step()
+def run(model_path: str, data_path: str = "", strategy: list = [], file_name = ""):
+    global trainset
+    global trainloader
+
+    if IN_COLAB:
+        model_path = "/content/drive/MyDrive/" + model_path
+
+    if data_path:
+        load_subset_dataset(trainloader, data_path)
+    else:
+        trainset = torchvision.datasets.CIFAR10(
+            root='./data', train=True, download=True, transform=IMAGE_PREPROCESSING)
+        trainloader = torch.utils.data.DataLoader(
+            trainset, batch_size=128, shuffle=True, num_workers=2)
+
+    if strategy and file_name:
+        reduce_dataset_over_strategy(trainloader, strategy, file_name)
+        load_subset_dataset(trainloader, "./scenarios/"+ file_name)
+
+    for epoch in range(start_epoch, start_epoch + 55):
+        train(epoch)
+        test(epoch, model_path)
+        scheduler.step()
+
+
+def test_model(model_path: str, data_path: str = "", strategy: list = [], file_name: str = ""):
```

Comment:
Tutaj raczej jedynym parametrem powinien być jedynie ścieżka do modelu (na razie, jak dojdą metody do redukcji niezbalansowania to będzie więcej :P).